cmd/kubeturbo/app/kubeturbo_builder.go

```diff
@@ -1044,14 +1048,43 @@ func WatchConfigMap() {
 				} else {
 					glog.V(1).Infof("Logging level is changed from %v to %v", currentLoggingLevel, newLoggingLevel)
 				}
-
 			}
 		}
+
+		updateNodePoolConfig(cluster.MinNodesConfigKey, &currentMinNodes, cluster.DefaultMinNodePoolSize, viper.GetString)
+		updateNodePoolConfig(cluster.MaxNodesConfigKey, &currentMaxNodes, cluster.DefaultMaxNodePoolSize, viper.GetString)
 	}
-	updateConfig() //update the logging level during startup
+	updateConfigClosure() //update the logging level during startup
 	viper.OnConfigChange(func(in fsnotify.Event) {
-		updateConfig()
+		updateConfigClosure()
 	})
 
 	viper.WatchConfig()
 }
+
+// updateNodePoolConfig updates the value of a configuration setting for an integer property.
+// If the configuration value is empty, it uses the provided defaultValue.
+// If the configuration value is not a valid integer or is negative, it uses the defaultValue and logs an error.
+// If the configuration value is different from the currentValue, it updates the currentValue and logs the change.
+func updateNodePoolConfig(configKey string, currentValue *int, defaultValue int, getKeyStringVal func(string) string) {
+	glog.V(1).Infof("Cluster %s current value is %v ", configKey, *currentValue)
+	newValueStr := getKeyStringVal(configKey)
+
+	if newValueStr == "" {
```

Comment:
Suppose the user ONLY defined `logging.level` in the configMap,  whenever the user updates the `logging.level`, this `("Empty value for %s in the configuration, using default value %d", configKey, defaultValue)` will be printed out according to the current logic.   You may need to make the similar change as what you did for the logging level.

Reply:
I think that is what we intended to do. If we do not specify the value for min and max of the node group, regardless of what the current values for min and max, we will revert that to the default.

Reply:
I think we should only print out message when there is actually an effective change. In most of the cases, there won't be settings for min/max node groups in the configMaps, then every time user changes a logging level, there will be a message printed out for min/max nodegroup which is not necessary and could actually be confusing.

Reply:
I think the solution is:
1) If the current min/max value is not default, we will print out the message of reverting to default if min/max were not specified in the configmap with message printed out
2) If the current min/max value is default, we will just do nothing with it. So no message will be seen over here.